pkg/ingester/flush.go

```diff
@@ -147,13 +147,13 @@ func (i *Ingester) shouldFlushSeries(series *memorySeries, immediate bool) flush
 	return noFlush
 }
 
-func (i *Ingester) shouldFlushChunk(c *desc) flushReason {
+func (i *Ingester) shouldFlushChunk(c *desc, fp model.Fingerprint) flushReason {
 	if c.flushed { // don't flush chunks we've already flushed
 		return noFlush
 	}
 
 	// Chunks should be flushed if they span longer than MaxChunkAge
-	if c.LastTime.Sub(c.FirstTime) > i.cfg.MaxChunkAge {
+	if c.LastTime.Sub(c.FirstTime) > (i.cfg.MaxChunkAge - time.Duration(fp)%i.cfg.ChunkAgeJitter) {
```

Comment:
If someone decides to turn off jitter (set it to 0), then this will panic. It would be nice to have some safety around that.

Reply:
Good point.